ts/parts-map/HeatmapSeries.ts

```diff
@@ -435,11 +739,50 @@ seriesType<Highcharts.HeatmapSeries>(
 
             // Get the extremes from the y data
             Series.prototype.getExtremes.call(this);
+        },
+
+        /**
+         * @private
+         * @borrows Highcharts.seriesTypes.line.setOptions as Highcharts.seriesTypes.heatmap#setOptions
+         */
+        setOptions: function (
```

Comment:
Mutating the options can be considered bad practice. Instead, this functionality should probably be moved to the `pointAttribs`, and the old options be applied directly to the attributes.

Reply:
Still an issue as of March 13.